streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -96,8 +96,10 @@ public void onCompletion(RecordMetadata metadata, Exception exception) {
                             TopicPartition tp = new TopicPartition(metadata.topic(), metadata.partition());
                             offsets.put(tp, metadata.offset());
                         } else {
-                            sendException = exception;
-                            log.error("{} Error sending record to topic {}. No more offsets will be recorded for this task and the exception will eventually be thrown", logPrefix, topic, exception);
+                            if (sendException == null) {
```

Comment:
we are changing it so it just reports the first exception. Previously it was reporting all of them, which for a batch of records was the same message how ever many times. In the case i was looking at it was a couple of hundred messages.